packages/core/src/createBYOKConfig.ts

```diff
@@ -0,0 +1,80 @@
+import invariant from 'tiny-invariant'
+import type { SignMessageReturnType } from 'viem'
+import type { BaseConfig } from './createConfig.js'
+import type * as rustUtils from './utils.d.ts'
+
+export type CreateBYOKConfigParameters = {
+  /** URL of the relayer service */
+  relayerUrl: string
+  /** URL of the websocket service */
+  websocketUrl: string
+  /**
+   * Function to sign messages using the user's own wallet.
+   *
+   * TODO: Move this to docs
+   * Important requirements for the signing function:
+   * 1. The function receives a pre-hashed message (using keccak256) - do not hash the message again
```

Comment:
why? this may not be something that all interfaces implement

Reply:
That's a fair point, we can just provide the message to be hashed and signed.